main.tf

```diff
@@ -1,132 +1,53 @@
-  trigger_prefixes          = var.trigger_prefixes
-  working_directory         = var.working_directory
-
-  dynamic "vcs_repo" {
-    for_each = local.connect_vcs_repo
-
-    content {
-      identifier         = var.repository_identifier
-      branch             = var.branch
-      ingress_submodules = false
-      oauth_token_id     = var.oauth_token_id
-    }
-  }
-}
-
-resource "tfe_workspace_settings" "default" {
-  agent_pool_id  = var.execution_mode == "agent" ? var.agent_pool_id : null
-  execution_mode = var.execution_mode
-  workspace_id   = tfe_workspace.default.id
-}
-
-resource "tfe_notification_configuration" "default" {
-  for_each = var.notification_configuration
-
-  name             = each.key
-  destination_type = each.value.destination_type
-  enabled          = each.value.enabled
-  triggers         = each.value.triggers
-  url              = each.value.url
-  workspace_id     = tfe_workspace.default.id
+module "tfe-workspace" {
+  source  = "schubergphilis/mcaf-workspace/tfe"
+  version = "2.0.0"
+
+  name                           = var.name
+  agent_pool_id                  = var.execution_mode == "agent" ? var.agent_pool_id : null
+  allow_destroy_plan             = var.allow_destroy_plan
+  assessments_enabled            = var.assessments_enabled
+  auto_apply                     = var.auto_apply
+  auto_apply_run_trigger         = var.auto_apply_run_trigger
+  branch                         = var.branch
+  clear_text_env_variables       = var.clear_text_env_variables
+  clear_text_hcl_variables       = var.clear_text_hcl_variables
+  clear_text_terraform_variables = var.clear_text_terraform_variables
+  description                    = var.description
+  execution_mode                 = var.execution_mode
+  file_triggers_enabled          = var.file_triggers_enabled
+  global_remote_state            = var.global_remote_state
+  notification_configuration     = var.notification_configuration
+  oauth_token_id                 = var.repository_identifier != null ? var.oauth_token_id : null
+  project_id                     = var.project_id
+  queue_all_runs                 = var.queue_all_runs
+  remote_state_consumer_ids      = var.remote_state_consumer_ids
+  repository_identifier          = var.repository_identifier
+  sensitive_env_variables        = var.sensitive_env_variables
+  sensitive_hcl_variables        = var.sensitive_hcl_variables
+  sensitive_terraform_variables  = var.sensitive_terraform_variables
+  ssh_key_id                     = var.ssh_key_id
+  terraform_organization         = var.terraform_organization
+  terraform_version              = var.terraform_version
+  trigger_patterns               = var.trigger_patterns
+  trigger_prefixes               = var.trigger_prefixes
+  variable_set_ids               = var.variable_set_ids
+  working_directory              = var.working_directory
+  workspace_tags                 = var.workspace_tags
 }
 
 resource "tfe_variable" "aws_default_region" {
```

Comment:
Not introduced in this PR, but maube we can merge this with the `var.clear_text_env_variables` list? Looks cleaner and prevents possible conflict (if somebody now passes `AWS_DEFAULT_REGION` it will error I think)